### DISCUSSION TITLE
Passing bytes to table

### DISCUSSION THREAD

**Discussion by genderev** [perspective-dev/perspective#1425] [Q&A] [answered]
## Support Question

I tried to copy an [example](https://github.com/finos/perspective/blob/master/examples/tornado-python/server.py) that passed bytes to a `Table()` and saw in the [Python documentation](https://perspective.finos.org/docs/obj/perspective-python.html#loaddata-options-1) that any given instance of a `Table()` can accept bytes. 

I'm not sure what kind of file the `Table()` would accept, however. I tried `table = Table("filename.parquet")` without success.

(I'm sure the errors produced are due to my own misunderstanding of what a parquet is, but I'm not sure what the proper type of file would be. )

**Comment by sc1f** [comment, marked as answer]
[Here](https://github.com/finos/perspective/blob/master/examples/tornado-python/server.py) is a good example of loading an Arrow from `str`. The ":obj:`bytes`|:obj:`str`" part of the API docs probably need to be cleared up a bit (along with formatting, it seems)—it does not refer to the ability to load arbitrary strings/bytes, but rather the ability to load _CSVs_ as strings and _Apache Arrow-formatted binaries_ in a byte format.

I am not super familiar with Parquet, but this is the code we use in our tests to [convert an Apache Arrow `Table` to `str`](https://github.com/finos/perspective/blob/master/python/perspective/perspective/tests/conftest.py):

```python
stream = pa.BufferOutputStream()
arrow_table = pyarrow.Table.from_batches([batch])
writer = pyarrow.RecordBatchStreamWriter(stream, arrow_table.schema)
writer.write_table(arrow_table)
writer.close()
table = perspective.Table(stream.getvalue().to_pybytes())
```

**Reply by genderev** [reply]
That file was helpful, thanks.
